### PR TITLE
Fix/on viewport change integrity

### PR DESCRIPTION
**Pull request by vdubr**
I got sometimes error caused "can not read property viewport.center[0] of undefined"

### CODE REVIEW

File: src/ReactLeafletMap/index.js [hunk 1]
```diff
@@ -87,20 +87,29 @@ class ReactLeafletMap extends React.PureComponent {
...
     onViewportChange(viewport) {
-        const center = {
-            lat: viewport.center[0],
-            lon: viewport.center[1]
-        };
-
-        const boxRange = viewUtils.getBoxRangeFromZoomLevel(viewport.zoom);
+        const change = {};
+        // viewport.center could be undefined
+        if(viewport && viewport.hasOwnProperty('center') && viewport.center) {
```
Comment (vlach1989) [inline src/ReactLeafletMap/index.js:92]:
```suggestion
        if(viewport && viewport.center) {
```

File: src/ReactLeafletMap/index.js [hunk 2]
```diff
@@ -87,20 +87,29 @@ class ReactLeafletMap extends React.PureComponent {
...
+            const center = {
+                lat: viewport.center[0],
+                lon: viewport.center[1]
+            };
+            change['center'] = center;
+        }
+        
+        // viewport.zoom could be undefined
+        if(viewport && viewport.hasOwnProperty('zoom') && Number.isFinite(viewport.zoom)) {
```
Comment (vlach1989) [inline src/ReactLeafletMap/index.js:101]:
```suggestion
        if(viewport && viewport.zoom) {
```

File: src/ReactLeafletMap/index.js (hunk 1 above)
Comment (vdubr) [reply src/ReactLeafletMap/index.js:92]: As in comment viewport.center can be undefined

File: src/ReactLeafletMap/index.js [hunk 3]
```diff
@@ -87,20 +87,29 @@ class ReactLeafletMap extends React.PureComponent {
...
+            const boxRange = viewUtils.getBoxRangeFromZoomLevel(viewport.zoom);
+            change['boxRange'] = boxRange;
+        }
 
         // TODO for IndexedVectorLayer rerender (see IndexedVectorLayer render method)
         let stateUpdate = {viewport};
 
         if (this.props.onViewChange) {
```
Comment (vlach1989) [inline src/ReactLeafletMap/index.js:109]: Test in change is not epmty for whole if/else?

File: src/ReactLeafletMap/index.js (hunk 2 above)
Comment (vdubr) [reply src/ReactLeafletMap/index.js:101]: As in comment viewport.zoom can be undefined

File: src/ReactLeafletMap/index.js (hunk 3 above)
Comment (vdubr) [reply src/ReactLeafletMap/index.js:109]: You mean dont call onViewChange if change is empty?

File: src/ReactLeafletMap/index.js (hunk 1 above)
Comment (vlach1989) [reply src/ReactLeafletMap/index.js:92]: Then viewport && viewport.center is falsey

File: src/ReactLeafletMap/index.js (hunk 3 above)
Comment (vdubr) [reply src/ReactLeafletMap/index.js:109]: From react-leaflet doc: onViewportChange: (viewport: {center: ?[number, number], zoom: ?number}) => void (optional): fired continuously as the viewport changes.

So every onViewportChange contains center or zoom change.

File: src/ReactLeafletMap/index.js (hunk 1 above)
Comment (vdubr) [reply src/ReactLeafletMap/index.js:92]: true

File: src/ReactLeafletMap/index.js (hunk 3 above)
Comment (vlach1989) [reply src/ReactLeafletMap/index.js:109]: props.onViewChange and onViewportChange are two different methods

**Pull request merged**